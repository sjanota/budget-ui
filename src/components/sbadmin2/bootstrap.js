export const Variant = {
  secondary: 'secondary',
  primary: 'primary',
  danger: 'danger',
  success: 'success',
  warning: 'warning',
  white: 'white',
};

export const Size = {
  sm: 'sm',
  lg: 'lg',
};
