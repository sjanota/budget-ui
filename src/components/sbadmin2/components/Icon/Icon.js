import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

export default function Icon({ icon, className, ...props }) {
  const classNames = classnames('fas', `fa-${icon}`, 'fa-fw', className);
  return <i className={classNames} {...props} />;
}

Icon.propTypes = {
  icon: PropTypes.string.isRequired,
  className: PropTypes.string,
};

Icon.Archive = 'archive';
Icon.Trash = 'trash-alt';
Icon.Edit = 'edit';
Icon.Plus = 'plus';
Icon.Minus = 'minus';
Icon.Save = 'save';
Icon.ClipboardCheck = 'clipboard-check';
Icon.ChevronUp = 'chevron-up';
Icon.ChevronDown = 'chevron-down';
