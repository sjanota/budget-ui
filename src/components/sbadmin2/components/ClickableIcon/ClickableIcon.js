import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import { Variant } from '../../bootstrap';

export default function ClickableIcon({
  icon,
  variant,
  className,
  onClick,
  ...props
}) {
  const classes = classnames(
    'btn',
    'bg-transparent',
    'border-0',
    'p-0',
    'mx-1',
    className,
    {
      [`text-${variant}`]: variant,
    }
  );

  function handleOnClick(e) {
    e.preventDefault();
    onClick && onClick(e);
  }

  return (
    <button className={classes} onClick={handleOnClick} {...props}>
      <FontAwesomeIcon icon={icon} />
    </button>
  );
}

ClickableIcon.propTypes = {
  icon: PropTypes.any.isRequired,
  className: PropTypes.string,
  onClick: PropTypes.func,
  variant: PropTypes.oneOf(Object.keys(Variant)).isRequired,
};
