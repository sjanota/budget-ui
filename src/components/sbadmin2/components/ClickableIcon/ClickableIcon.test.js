import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import ClickableIcon from './ClickableIcon';
import Icon from '../Icon/Icon';
import { Variant } from '../../bootstrap';

it('ClickableIcon renders with minimal properly', () => {
  const component = renderer.create(
    <ClickableIcon icon={Icon.Archive} variant={Variant.primary} />
  );
  expect(component).toMatchSnapshot();
});

describe('ClickableIcon', () => {
  const component = mount(
    <ClickableIcon icon={Icon.Archive} variant={Variant.primary} />
  );

  it('sets proper text-variant class', () => {
    expect(component.find('button')).toHaveClassName('text-primary');
  });

  it('passes icon to Icon', () => {
    expect(component.find(Icon)).toHaveProp('icon', Icon.Archive);
  });
});
