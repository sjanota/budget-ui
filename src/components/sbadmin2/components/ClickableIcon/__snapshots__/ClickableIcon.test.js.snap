// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickableIcon renders with minimal properly 1`] = `
<button
  className="btn bg-transparent border-0 p-0 mx-1 text-primary"
>
  <i
    className="fas fa-archive fa-fw"
  />
</button>
`;
